### PR TITLE
Develop with latest core master

### PR DESCRIPTION
**Pull request by phil-davis**
Add some words to the developer docs intro pages so that readers will understand that the developer docs are intended for use with current core master. (There might be all sorts of different stuff if someone checks out an old release branch and tries to develop or run tests from those)

### CODE REVIEW

File: modules/developer_manual/pages/core/index.adoc [hunk 1]
```diff
@@ -1,3 +1,5 @@
 = Core Development
 
 In this section you will find all the details you need to develop ownCloud’s core.
+
+You should develop starting with the latest code in the core master branch. The core development documentation is written assuming that you are doing that.
```
Comment (mmattel) [inline modules/developer_manual/pages/core/index.adoc:5]:
```suggestion
IMPORTANT: You should develop starting with the latest code in the core master branch. The core development documentation is written assuming that you are doing that.
```
Not a must highlighting with important, but creates good visibility. Reject if not ok

**Review by mmattel — APPROVED** [review]

File: modules/developer_manual/pages/core/index.adoc (hunk 1 above)
Comment (phil-davis) [reply modules/developer_manual/pages/core/index.adoc:5]: IMO this is just a note/recommendation/information aimed at developers. So not really necessary to write "IMPORTANT".

**Comment by mmattel** [comment]
Post thoughts regarding backports. Would´nt it be good to at least BP to 10.6 (as 10.5 will be phased out soon due to upcoming 10.7) and someone who is reading "accidentially" 10.6 should get optical notified?

**Comment by phil-davis** [comment]
Yes, it only takes a minute! Backports:
10.6 #3285 
10.5 #3286

**Pull request merged**